xwiki-platform-core/xwiki-platform-realtime/xwiki-platform-realtime-default/src/main/java/org/xwiki/realtime/DefaultRealtimeEditorManager.java

```diff
@@ -0,0 +1,152 @@
+        }
+        if (requestEditor.length() > 0) {
+            return requestEditor;
+        }
+
+        // The inplace editor comes with a custom InplaceEditing sheet that handles the locking confirmation.
+        // to handle this special case, and potential future others, we add the possibility to set the selected
+        // editor through the editor variable in the ScriptContext.
+        String scontextEditor = "";
+        Object scontextEditorObj = this.scriptContextManager.getCurrentScriptContext().getAttribute(editorKey);
+        if (scontextEditorObj instanceof String) {
+            scontextEditor = (String) scontextEditorObj;
+        }
+        if (scontextEditor.length() > 0) {
+            return scontextEditor;
+        }
+
+        // Otherwise, we fallback to the default editor:
+        // This part is taken from the getDefaultDocumentEditor macro
+        // defined in macros.vm from xwiki-platform-web-templates.
+
+        // If a sheet matches the edit action for this document and no specific editor was specified,
+        // the inline form editor will be used.
+        XWikiContext context = (XWikiContext) this.xwikiContextProvider.get();
+        XWikiDocument document = (XWikiDocument) context.get("tdoc");
+        if (!this.sheetManager.getSheets(document, context.getAction()).isEmpty()) {
+            return "inline";
+        }
+
+        // If the default editor is set to Wysiwyg, it will be used if possible.
+        String xwikiEditorPreference = context.getWiki().getEditorPreference(context);
+        if (xwikiEditorPreference.equals(DefaultRealtimeEditorManager.wysiwyg)
+            && ((WysiwygEditorScriptService) wysiwygEditorScriptService)
+                .isSyntaxSupported(document.getSyntax().toIdString())) {
+            return xwikiEditorPreference;
+        }
+
+        return "wiki";
+    }
+
+    /**
+     * {@inheritDoc}
+     * 
+     * @see org.xwiki.realtime.RealtimeEditorManager#sessionIsActive(org.xwiki.model.reference.DocumentReference,
+     *      java.lang.String, java.lang.String)
+     */
+    public Boolean sessionIsActive(DocumentReference target, String locale, String editor)
+    {
+        // The inplace and WYSIWYG realtime editors both use the same "wysiwyg" channel.
+        String session = editor;
+        if (session.equals("inplace")) {
+            session = DefaultRealtimeEditorManager.wysiwyg;
+        }
+
+        // We can't directly specify the path because a document might have multiple fields.
+        // Instead, we check all the channels matching the given editor.
+        List<EntityChannel> channels = this.entityChannelStore.getChannels(target);
+        for (EntityChannel channel : channels) {
+            List<String> path = channel.getPath();
+            if (!path.isEmpty() && (path.get(path.size() - 1)).equals(session) && (path.get(0)).equals(locale)
+                && channel.getUserCount() > 0) {
```

Comment:
The if condition is hard to read, would be better for maintainability to break it with variables to help the reader :)